src/ParcelRegistry.Consumer.Address/Infrastructure/Program.cs

```diff
@@ -73,48 +78,63 @@ public static async Task Main(string[] args)
 
-                            var kafkaOffset = await GetOffset(container, loggerFactory.CreateLogger<Program>(), topic);
+                            var backOfficeConsumer = new BackOfficeConsumer(
+                                lifetimeScope,
+                                loggerFactory,
+                                kafkaOptions,
+                                topic,
+                                configuration["BackOfficeConsumerGroupSuffix"],
+                                backOfficeConsumerOffset);
+                            var backOfficeConsumerTask = backOfficeConsumer.Start(cancellationToken);
 
-                            await using (var consumerContext = actualContainer.Resolve<ConsumerAddressContext>())
-                            {
-                                var consumer = new Consumer(consumerContext, loggerFactory, kafkaOptions, topic,
-                                    consumerGroupSuffix, kafkaOffset);
-                                var consumerTask = consumer.Start(cancellationToken);
+                            var consumerTasks = new List<Task> {backOfficeConsumerTask};
 
-                                Log.Information("The kafka consumer was started");
+                            Log.Information("The kafka BackOfficeConsumer has started");
 
-                                await Task.WhenAny(consumerTask);
+                            var enableCommandHandlingConsumer = configuration["FeatureToggles:EnableCommandHandlingConsumer"];
+                            if (enableCommandHandlingConsumer != null && bool.Parse(enableCommandHandlingConsumer))
+                            {
+                                var commandHandlingConsumer = new CommandHandlingConsumer(
+                                    lifetimeScope,
+                                    loggerFactory,
+                                    kafkaOptions,
+                                    topic,
+                                    configuration["CommandHandlingConsumerGroupSuffix"]);
+                                var commandHandlingConsumerTask = commandHandlingConsumer.Start(cancellationToken);
+                                consumerTasks.Add(commandHandlingConsumerTask);
+                                
+                                Log.Information("The kafka CommandHandlingConsumer has started");
+                            }
 
-                                CancellationTokenSource.Cancel();
+                            await Task.WhenAny(consumerTasks);
 
-                                Log.Error($"Consumer task stopped with status: {consumerTask.Status}");
-                            }
+                            CancellationTokenSource.Cancel();
 
-                            Log.Error("The consumer was terminated");
+                            Log.Error("The Address Consumers were terminated");
```

Comment:
Can this be looped? It's interesting to know which one crashed (faulted state) and which ones were cancelled.

Reply:
Still need to do this